General Improvements rev 2 (#61)
matter-labs/foundry-zksync-book: src/zksync-specifics/gas.md

### Gas reported back to the EVM
Foundry has an `isolate` mode for the EVM, in which all `CALL`/`CREATE` operations at the root level of a test (i.e., with depth 1) are intercepted and treated as independent transactions. This allows for accounting for the actual transaction gas, including, for example, the fixed 21000 gas cost charged to the user.

Running in zkEVM mode is analogous to running in `isolate` mode, but using the zkEVM instead. Here's a short summary:

> - Every `CALL`/`CREATE` will be intercepted, a transaction representing the operation built. 

Reviewer: ```suggestion
- Every `CALL`/`CREATE` will be intercepted and a zkEVM transaction representing the operation built. 
```

- A VM with that transaction in the bootloader's heap will be spawned and run to simulate the execution of that transaction.

> - The gas used is reported back to the EVM; hence, the one seen on traces and gas reports would be charged to the user for submitting that transaction. That value differs from the computational cost of running the called contract code and includes: 

Reviewer: ```suggestion
- The gas used is reported back to the EVM so it will then be shown on traces and gas reports. This value represents what would be charged to the user for submitting that transaction. That value differs from the computational cost of running the called contract code and includes: 
```
